src/interpreter_funcs/boolean_intersection.rs

```diff
@@ -0,0 +1,123 @@
+use std::error;
+use std::f32;
+use std::fmt;
+use std::sync::Arc;
+
+use nalgebra::Vector3;
+
+use crate::interpreter::{
+    Float3ParamRefinement, Func, FuncError, FuncFlags, FuncInfo, ParamInfo, ParamRefinement, Ty,
+    UintParamRefinement, Value,
+};
+use crate::mesh::voxel_cloud::VoxelCloud;
+
+#[derive(Debug, PartialEq)]
+pub enum FuncBooleanIntersectionError {
+    WeldFailed,
+    NotIntersecting,
```

Comment:
Instead of this, I'd change this to the more general error case when the resulting voxel cloud is empty. If you want to keep to give the user more info, that's fine as well, but in that case we need another error variant, because the interesection can be empty even if the voxel clouds intersect.